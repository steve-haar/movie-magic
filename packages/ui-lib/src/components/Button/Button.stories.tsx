import type { Meta, StoryObj } from '@storybook/react';
import { Button } from './Button';

const meta = {
  title: 'Components/Button',
  component: Button,
  tags: ['autodocs'],
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

export const DefaultButton: Story = {
  args: {
    children: 'Button',
  },
};

export const PrimaryButton: Story = {
  args: {
    color: 'primary',
    children: 'Button',
  },
};

export const SecondaryButton: Story = {
  args: {
    color: 'secondary',
    children: 'Button',
  },
};
