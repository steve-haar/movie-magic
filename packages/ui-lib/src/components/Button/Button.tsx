import * as React from 'react';

type ReactButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement>;

interface ButtonProps extends ReactButtonProps {
  color?: 'default' | 'primary' | 'secondary';
}

export function Button({
  className,
  color = 'default',
  children,
  onClick,
}: ButtonProps) {
  const classes = [];

  if (className) {
    classes.push(className);
  }

  classes.push('button button--contained button--small');

  switch (color) {
    case 'default': {
      classes.push('button--contained-default');
      break;
    }
    case 'primary': {
      classes.push('button--contained-primary');
      break;
    }
    case 'secondary': {
      classes.push('button--contained-secondary');
      break;
    }
  }

  return (
    <button className={classes.join(' ')} onClick={onClick}>
      {children}
    </button>
  );
}
