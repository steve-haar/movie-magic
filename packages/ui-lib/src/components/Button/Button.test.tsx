import { Button, buttonVariants } from './Button';
import { render } from '../../test/test-utils';

describe('<Button />', () => {
  it('should render correct default properties', () => {
    const { asFragment } = render(<Button>Submit</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correct variants', () => {
    const { asFragment } = render(
      <div>
        {buttonVariants.map((variant) => (
          <Button key={variant} variant={variant}>
            Submit
          </Button>
        ))}
      </div>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should allow to add a class using className prop', () => {
    const { asFragment } = render(<Button className="ml-2">Submit</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
