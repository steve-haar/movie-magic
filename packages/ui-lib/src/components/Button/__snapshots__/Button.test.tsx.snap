// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> class can be added via rootClass prop 1`] = `
<DocumentFragment>
  <button
    class="ml-2 button button--default"
  >
    Submit
  </button>
</DocumentFragment>
`;

exports[`<Button /> colors render correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      class="button button--default"
    >
      Submit
    </button>
    <button
      class="button button--primary"
    >
      Submit
    </button>
    <button
      class="button button--secondary"
    >
      Submit
    </button>
  </div>
</DocumentFragment>
`;
